### PR TITLE
feat(#5864): Added configuration for sizeLimit on empty-dirs in mount trait

### PR DESCRIPTION
**Pull request by hernanDatgDev**
The default SizeLimit on mount trait empty-dir volumes is now configurable with the following syntax:
trait.mount. empty-dirs=myVolumeName:/container/path:sizeLimit

Where sizeLimit is a normal resource quantity e.g. 1Gi, 500Mi, etc.

I considered something similar to components where we add configurations after a "?" however since the empty-dir volume source only takes 2 configurations: 
SizeLimit
Medium

I figured it would be easier to just continue the ":" separation syntax. Please let me know your thoughts on this approach

**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Comment by hernanDatgDev** [comment]
I need to update the documentation for the syntax on the trait still unless you have any concerns with the approach and I should change it. @squakez

**Review by squakez — COMMENTED** [review]
I think the parsing is getting too complicated and out of the original scope. Probably it makes sense that we have a separate section in the mount trait that takes care to parse and to add volumes coming from the empty-dir parameter independently by the rest. In that way we can treat them appropriately adding any possible future syntax change.

Beside that, it would be good to set a default value on resource when the user is not providing it (either 1GB or anything lower).

**Comment by hernanDatgDev** [comment]
Made the changes so that logic is contained mostly within the mount trait. Parsing is kept to a minimum and I'd just need to update the syntax documentation if the logic looks good @squakez

**Review by squakez — APPROVED** [review]
LGTM. Please add the doc changes and have a look at the lint failure.

**Comment by hernanDatgDev** [comment]
@squakez I've run `make generate` to handle the documentation update but there are a lot of additional changes that result from this change, specifically to the CRD definitions. They updates do include my updated documentation though.

Any thoughts? I can always revert this commit.

File: pkg/resources/config/crd/bases/camel.apache.org_builds.yaml [hunk 1]
```diff
@@ -15,12 +15,14 @@
...
 # ---------------------------------------------------------------------------
 
+
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.15.0
+    controller-gen.kubebuilder.io/version: v0.6.1
```
Comment (squakez) [inline pkg/resources/config/crd/bases/camel.apache.org_builds.yaml:24]: I think the problem is the controller gen version used. You may need to upgrade to a more recent one.

**Review by squakez — COMMENTED** [review]
I see you're using an old controller-gen version. You need to upgrade to a more recent one (likely 0.15.0) and redo the generation.

**Comment by squakez** [comment]
Hi @hernanDatgDev how this one is going? I am planning to release 2.5.0 for next week, so, we should have this ready by then to include in the release.

**Comment by hernanDatgDev** [comment]
@squakez I'm sorry about the delay. I will get this updated and ready as soon as I can.

File: pkg/resources/config/crd/bases/camel.apache.org_builds.yaml (hunk 1 above)
Comment (hernanDatgDev) [reply pkg/resources/config/crd/bases/camel.apache.org_builds.yaml]: @squakez I'm running `make generate` while the version is 0.15.0 but seems to be getting overridden somewhere after the generate command completes. I feel like I'm still missing something but not sure what yet. I synced my code and resolved conflicts too. Is there a new method that's being used for code generation?

**Comment by squakez** [comment]
Let's wait for the checks to run. For now, don't worry about the autogenerated documentation, we have a nightly task taking care to update. However, for the future, we need to take care of synchronizing the tooling automatically, so, I'm opening a separate issue to take care of that.

**Comment by squakez** [comment]
`pkg/trait/mount.go:284:65  godot  Comment should end in a period`

Please, fix and `make lint` locally to run the validation and see no more errors.

**Pull request merged**